### ISSUE TITLE
`dotnet list package --outdated` lists up to date packages on `.Net 6.0.100-preview.7.21364.30`

### ISSUE THREAD

**Issue by wli3** [NuGet/Home#11139]
---
**Issue moved from dotnet/sdk#18942**
- Please respond to @MichalPetryka.

---

_From @MichalPetryka on Thursday, July 15, 2021 3:49:06 PM_

For a project that has any nuget packages installed, `dotnet list package --outdated` lists all the packages as updatable, reporting the same version as both the currently installed and the newest one. On .Net 5 packages which were on the latest version weren't listed, which I assume is the correct behaviour. For a project without any nuget packages, the output correctly says that no updates are available.

Reproduction:
```
dotnet new xunit -o Test
cd Test
dotnet restore
dotnet list package --outdated
```

Expected output:
```
> coverlet.collector             3.0.2    3.0.2        3.0.3
> Microsoft.NET.Test.Sdk         16.9.4   16.9.4       16.10.0
```

Actual output:
```
> coverlet.collector             3.0.2    3.0.2        3.0.3
> Microsoft.NET.Test.Sdk         16.9.4   16.9.4       16.10.0
> xunit                          2.4.1    2.4.1        2.4.1
> xunit.runner.visualstudio      2.4.3    2.4.3        2.4.3
```

**Comment by wli3** [comment]
---
**Issue moved from dotnet/sdk#18942**
- Please respond to @dotnet-issue-labeler[bot].

---

_From @dotnet-issue-labeler[bot] on Thursday, July 15, 2021 3:49:08 PM_

I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by erdembayar** [comment]
@aortiz-msft 
I can repro this regression.

dotnet sdk 5.100
C:\NuGetProj\IssueRepro\11139\Test> dotnet list package --outdated

The following sources were used:
   https://api.nuget.org/v3/index.json
   C:\NuGetProj\Mypackages
   https://pkgs.dev.azure.com/devdiv/_packaging/vs-corextfeeds/nuget/v3/index.json

Project `Test` has the following updates to its packages
   [net5.0]:
   Top-level Package             Requested   Resolved   Latest
   > coverlet.collector          3.0.2       3.0.2      3.1.0
   > Microsoft.NET.Test.Sdk      16.9.4      16.9.4     16.10.0

C:\NuGetProj\IssueRepro\11139\Test> dotnet --version
6.0.100-preview.7.21377.35

C:\NuGetProj\IssueRepro\11139\Test> dotnet list package --outdated

The following sources were used:
   https://api.nuget.org/v3/index.json
   C:\NuGetProj\Mypackages
   https://pkgs.dev.azure.com/devdiv/_packaging/vs-corextfeeds/nuget/v3/index.json

Project `Test` has the following updates to its packages
   [net5.0]:
   Top-level Package                Requested   Resolved   Latest
   > coverlet.collector             3.0.2       3.0.2      3.1.0
   > Microsoft.NET.Test.Sdk         16.9.4      16.9.4     16.10.0
   > xunit                          2.4.1       2.4.1      2.4.1
   > xunit.runner.visualstudio      2.4.3       2.4.3      2.4.3

**Comment by erdembayar** [comment]
See https://github.com/NuGet/Home/issues/11137#issuecomment-897345373, they're connected.

**Comment by aortiz-msft** [comment]
@drewgillies - Same as the issue for deprecated.

cc @joelverhagen

**Comment by drewgillies** [comment]
Note that there was a profound regression (breaks in filtering for all non-generic reports) in https://github.com/NuGet/NuGet.Client/pull/3699, fixed in https://github.com/NuGet/NuGet.Client/pull/4158. This latter change seems well after the release being reported on here, and appears not to be inserted in dotnet/sdk yet.
/cc @joelverhagen @aortiz-msft @wli3

**Comment by aortiz-msft** [comment]
Fixed in VS 2022 Preview 4; NET 6.0 RC1.
